Unity-Technologies/com.unity.multiplayer.samples.bitesize
# Sample: API diorama

[style guide]: https://unity-docs.gitbook.io/style-guide/
[release note guide lines]: https://unity-docs.gitbook.io/style-guide/style/release-notes
[coding conventions]: https://github.cds.internal.unity3d.com/unity/unity-meta/blob/stable/ReferenceSource/CSharp/Assets/CSharpReference.cs


### Proposed change(s)

1. Added the foundation of a new sample named "API Diorama" (details in the JIRA ticket linked below)
2. Added a Pull request template
3. All Parrelsync's "clone" folder of the whole repo are now ignored

### Platform and Editor version used to develop and test this PR
- Windows, 2021.3.12f1

### Related links
- [JIRA](https://jira.unity3d.com/browse/MTT-5094)

### Affected sample(s)

- [ ] 2DSpaceShooter
- [ ] Invaders
- [ ] Client Driven
- [x] API Diorama
- [ ] N/A (please describe)

### Types of change(s)

- [ ] UI/UX (see [style guide] and add Jil as a reviewer for relevant changes)
- [ ] Documentation update (see [style guide])
- [x] Other (please describe) --> new sample

### Checklist
- [ ] Unit tests pass locally --> n/a right now
- [ ] Added tests, if feasible, that prove my fix is effective or that my feature works 
- [x] No warnings in the Console
- [x] Documented (///) **public APIs** (public/protected functionality of public/protected classes), if applicable
- [x] Updated the sample documentation, if applicable (see [style guide])
- [ ] UI & Style sheet changes: both Dark and Light styles/themes updated/tested --> n/a right now
- Remember to familiarize yourself with the [coding conventions]

### How to test/validate

1. Clone this branch
2. Open it with the aforementioned Unity version
4. Ensure the Welcome Dialog pops up when you open the project for the first time (you can open it again at any time from Tutorials > Show welcome dialog)
5. Click "Start tutorials"
6. Check that the tutorial starts
7. Complete it (this will make you go through all the acceptance criteria of the linked JIRA tickets)
8. Open the `SelectionScreen` scene
9. Press "Play"
10. Ensure that a single image for each available sample scene is loaded in the center of the screen, and that clicking it will open the associated scene
11. Ensure that the links of the tutorial's "table of content" work (you can open it from Tutorials > Show tutorials)

### Other comments

1. For some reason, the light of the "NetworkVariable vs RPC" scene are "broken" in the editor, when the scene is loaded through the scene selection scene. Opening the scene directly or building the game will result in correct lighting.
2. The bundled scene is just an example. We can add more scenes at any time to explain new feature.
3. UI images and scene art is temporary and can be improved at any time in a separate PR.
